datamodel.customer-survey.xml

```diff
@@ -1922,6 +1925,56 @@ EOF
 				throw new Exception(Dict::S('Survey-email-notsent').': '.implode(', ', $aErrors));
 		}
 	}]]></code>
+        </method>
+        <method id="ApplyParamsToContent">
+          <comment>/**
+            * @param string $sContent
+            * @param Contact $oContact
+            * @param string $sQuizUrl
+            * @param boolean $bAutoAppendLink if true will call {@see AppendLinkToContent}
+            *
+            * @return string Content with placeholders replaced.
+            */</comment>
+          <static>false</static>
+          <access>protected</access>
+          <type>Custom</type>
+          <code><![CDATA[	protected function ApplyParamsToContent($sContent, $oContact, $sQuizUrl, $bAutoAppendLink)
+{
+  $aParams = [];
+  $aParams['recipient->object()'] = $oContact;
+  $aParams['quiz_url'] = $sQuizUrl;
+  $aParams['this->object()'] = $this;
+
+  if ($bAutoAppendLink){
+    $sContent = $this->AppendLinkToContent($sContent, $sQuizUrl);
+  }
+
+  $sContent = MetaModel::ApplyParams($sContent, $aParams);
+
+  return $sContent;
+}
+]]></code>
+        </method>
+        <method id="AppendLinkToContent">
+          <comment>/**
+            * @param string $sContent
+            * @param string $sQuizUrl
+            *
+            * @return string If a placeholder is found in $sContent returns $sContent, otherwise $sContent with a quizz link appended
+            */</comment>
+          <static>true</static>
+          <access>public</access>
+          <type>Custom</type>
+          <code><![CDATA[	public static function AppendLinkToContent($sContent, $sQuizUrl)
+{
+    $quizUrlPatternFound = preg_match('/\$quiz_url\$/', $sContent);
```

Comment:
We may need to revisit this. Is the $ sign enough here?
I think sometimes the encoded version (%24 instead of $) is passed. So we'd need to adjust this regex.

Reply:
Indeed, you can find a similar behavior in `\MetaModel::ApplyParams()`